[Web] Can't install onnxruntime-node behind corporate proxy

### Describe the issue

```
npm ERR! command failed
npm ERR! command sh -c node ./script/install
npm ERR! Downloading "https://github.com/microsoft/onnxruntime/releases/download/v1.20.1/onnxruntime-linux-x64-gpu-1.20.1.tgz"...
npm ERR! node:internal/deps/undici/undici:11576
npm ERR!     Error.captureStackTrace(err, this);
npm ERR!           ^
npm ERR!
npm ERR! TypeError: fetch failed
npm ERR!     at Object.fetch (node:internal/deps/undici/undici:11576:11)
npm ERR!     at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
npm ERR!   cause: ConnectTimeoutError: Connect Timeout Error
npm ERR!       at onConnectTimeout (node:internal/deps/undici/undici:8522:28)
npm ERR!       at node:internal/deps/undici/undici:8480:50
npm ERR!       at Immediate._onImmediate (node:internal/deps/undici/undici:8511:13)
npm ERR!       at process.processImmediate (node:internal/timers:476:21) {
npm ERR!     code: 'UND_ERR_CONNECT_TIMEOUT'
npm ERR!   }
npm ERR! }
npm ERR!
npm ERR! Node.js v18.17.0
```

The `fetch` doesn't respect http_proxy and https_proxy set in the environment:
https://github.com/microsoft/onnxruntime/blob/86870114eb14c459533034a1dd6f9b722cea0f9f/js/node/script/install.js#L83


### To reproduce

```
npm install onnxruntime-node
```

### Urgency

_No response_

### ONNX Runtime Installation

Other / Unknown

### ONNX Runtime Version or Commit ID

1.20.1

### Execution Provider

Other / Unknown

--- a/js/node/package.json
+++ b/js/node/package.json
@@ -15,6 +15,7 @@
   },
   "version": "1.21.0",
   "dependencies": {
+    "global-agent": "^3.0.0",
     "onnxruntime-common": "file:../common",
     "tar": "^7.0.1"
   },

--- a/js/node/script/install.js
+++ b/js/node/script/install.js
@@ -19,6 +19,7 @@
 // Step.1: Check if we should exit early
 const os = require('os');
 const fs = require('fs');
+const https = require('https');
 const path = require('path');
 const tar = require('tar');
 const { execFileSync } = require('child_process');
@@ -80,44 +81,62 @@ To use ONNX Runtime Node.js binding with CUDA v11 support, please follow the man
   }.tgz`,
 }[INSTALL_CUDA_FLAG || tryGetCudaVersion()];
 console.log(`Downloading "${artifactUrl}"...`);
-fetch(artifactUrl).then((res) => {
-  if (!res.ok) {
-    throw new Error(`Failed to download the binaries: ${res.status} ${res.statusText}.
+
+const FILES = new Set([
+  'libonnxruntime_providers_tensorrt.so',
+  'libonnxruntime_providers_shared.so',
+  `libonnxruntime.so.${ORT_VERSION}`,
+  'libonnxruntime_providers_cuda.so',
+]);
+
+downloadAndExtract(artifactUrl, BIN_FOLDER, FILES);
+
+async function downloadAndExtract(url, dest, files) {
+  return new Promise((resolve, reject) => {
+    https.get(url, (res) => {
+      const { statusCode } = res;
+      const contentType = res.headers['content-type'];
+
+      if (statusCode === 301 || statusCode === 302) {
+        downloadAndExtract(res.headers.location, dest, files).then(
+          (value) => resolve(value),
+          (reason) => reject(reason),
+        );
+        return;
+      } else if (statusCode !== 200) {
+        throw new Error(`Failed to download the binaries: ${res.statusCode} ${res.statusMessage}.
 
 Use "--onnxruntime-node-install-cuda=skip" to skip the installation. You will still be able to use ONNX Runtime, but the CUDA EP will not be available.`);
-  }
+      }
 
-  // Extract the binaries
+      if (!contentType || !/^application\/octet-stream/.test(contentType)) {
+        throw new Error(`unexpected content type: ${contentType}`);
+      }
 
-  const FILES = new Set([
-    'libonnxruntime_providers_tensorrt.so',
-    'libonnxruntime_providers_shared.so',
-    `libonnxruntime.so.${ORT_VERSION}`,
-    'libonnxruntime_providers_cuda.so',
-  ]);
-
-  Readable.fromWeb(res.body)
-    .pipe(
-      tar.t({
-        strict: true,
-        onentry: (entry) => {
-          const filename = path.basename(entry.path);
-          if (entry.type === 'File' && FILES.has(filename)) {
-            console.log(`Extracting "${filename}" to "${BIN_FOLDER}"...`);
-            entry.pipe(fs.createWriteStream(path.join(BIN_FOLDER, filename)));
-            entry.on('finish', () => {
-              console.log(`Finished extracting "${filename}".`);
-            });
-          }
-        },
-      }),
-    )
-    .on('error', (err) => {
-      throw new Error(`Failed to extract the binaries: ${err.message}.
+      res
+        .pipe(
+          tar.t({
+            strict: true,
+            onentry: (entry) => {
+              const filename = path.basename(entry.path);
+              if (entry.type === 'File' && files.has(filename)) {
+                console.log(`Extracting "${filename}" to "${dest}"...`);
+                entry.pipe(fs.createWriteStream(path.join(dest, filename)));
+                entry.on('finish', () => {
+                  console.log(`Finished extracting "${filename}".`);
+                });
+              }
+            },
+          }),
+        )
+        .on('error', (err) => {
+          throw new Error(`Failed to extract the binaries: ${err.message}.
 
 Use "--onnxruntime-node-install-cuda=skip" to skip the installation. You will still be able to use ONNX Runtime, but the CUDA EP will not be available.`);
+        });
     });
-});
+  });
+}
 
 function tryGetCudaVersion() {
   // Should only return 11 or 12.

--- a/js/node/script/install.js
+++ b/js/node/script/install.js
@@ -23,7 +23,12 @@ const https = require('https');
 const path = require('path');
 const tar = require('tar');
 const { execFileSync } = require('child_process');
-const { Readable } = require('stream');
+const { bootstrap: globalAgentBootstrap } = require('global-agent');
+
+// Bootstrap global-agent to honor the proxy settings in
+// environment variables, e.g. GLOBAL_AGENT_HTTPS_PROXY.
+// See https://github.com/gajus/global-agent/blob/v3.0.0/README.md#environment-variables for details.
+globalAgentBootstrap();
 
 // commandline flag:
 // --onnxruntime-node-install-cuda         Force install the CUDA EP binaries. Try to detect the CUDA version.

